### ISSUE TITLE
Double check errors API in client-go 10

### ISSUE THREAD

**Issue by hausdorff** [pulumi/pulumi-kubernetes#360]
During the client-go v10 upgrade (#348) we realized that the error types returned in various places have changed. _e.g._, it seems that when a CRD is not found, it no longer returns a 404, and instead returns a new error.

This is scary, and we should audit the error handling to make sure we're not missing something. Context: https://github.com/pulumi/pulumi-kubernetes/pull/348#discussion_r249188717

**Comment by lblackstone** [comment]
I took another look at the refactored code, and the only difference I found was here: https://github.com/pulumi/pulumi-kubernetes/blob/fb8073fbd2887cfc2f0288ebd21fe54bfcff1e9f/pkg/await/await.go#L97-L123

It appears that the difference is due to the caching layer [1] in the dynamic client, but I believe it is correct, and doesn't alter the semantics.

[1] As part of #348, the downstream cached client was removed in favor of the upstream implementation. The behavior is mostly identical, but the upstream client will return `IsNoMatchError` errors in case discovery fails for a particular Kind, as can happen when a group CRDs/CRs are applied out of order.
